README.md

```diff
@@ -1,2 +1,118 @@
-# strigo-github-exercises-examples
-Examples for markdown exercises that may be used in Strigo training events
+# Strigo GitHub Class Exercises examples
+
+## Quick start
+
+1. Create `.strigo/config.yml` file in your exercises' repository
+1. Configure exercises in `.strigo/config`
+
+   ```yaml
+   ---
+   classes:
+     - exercises:
+         # `file` is relative to .strigo directory
+         - file: "../Exercise.md"
+           title: "Title!"
+   ```
+
+1. In Strigo Class editor, fetch Exercises for a desired Class
+1. Done!
+
+---
+
+## Config file structure
+
+- `classes` - array of Classes
+  - `class_id` - defines the Class ID, optional if the configuration has only one Class
+  - `exercises` - array of Class' Exercises, cannot be empty
+    - `file` - path to Exercise markup file, relative to `.strigo` directory, supported extensions `.md` (markdown) and `.rst` (restructured), any other extension should be overriden by `syntax` field (see below)
+    - `title` - defines Exercise title in Strigo interface
+    - `syntax` - overrides `file` extension, supported syntaxes `.md` (markdown) and `.rst` (restructured)
+
+---
+
+## Main concepts
+
+1. **Hello world**
+
+   Just configure the order of the files, set titles and it is good to go.
+
+   ```yaml
+   ---
+   classes:
+     - exercises:
+         # `file` is relative to .strigo directory
+         - file: "../hello-world/hello.md"
+           title: "Hello!"
+         - file: "../hello-world/world.txt"
+           title: "World!"
+   ```
+
+1. **Multiple Classes in a single repository**
+
+   You can have multiple Classes in a single repository, defining Class IDs for each.
+
+   ```yaml
+   ---
+   classes:
+     - class_id: "<class_id_1>"
+       exercises:
+         - file: "../1.md"
+           title: "Title 1!"
+         - file: "../2.md"
+           title: "Title 2!"
+     - class_id: "<class_id_2>"
+       exercises:
+         - file: "../3.md"
+           title: "Title 3!"
+         - file: "../4.md"
+           title: "Title 4!"
+   ```
+
+1. **Reusing Exercises for different classes**
+
+   There is no restriction on using the same file for multiple Classes, or even using the same file multiple times for the same Class
+
+   ```yaml
+   ---
+   classes:
```

Comment:
I'd consider YAML shortcuts here, like in https://github.com/strigo/taco/blob/master/.circleci/config.yml#L61

Reply:
Prefer not to. In this way it is easily copy-pasteable.

Reply:
Well, it's copy-pastable even if you use shortcuts. You repeat the same two lines (title, file) multiple times.

Reply:
That's exactly the point of this example, it shows that you can reuse the same file many times. I prefer not to make it too complicated, I am assuming that not everyone is familiar with YAML, and think this should be done in the simplest possible way.